### PR TITLE
New cross region replication mm2 converter

### PR DESCRIPTION
**Pull request by JC-CJ2222**
Fixed CR: Modified Converter, Deleted Redundant Changes and Unused Dependency

### CODE REVIEW

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 1]
```diff
@@ -186,15 +208,29 @@ private void validateAndSetAWSEndpoint(Map<String, ?> configs) {
...
     private void validateAndSetProxyUrl(Map<String, ?> configs) {
         if (isPresent(configs, AWSSchemaRegistryConstants.PROXY_URL)) {
-    		String value = (String) configs.get(AWSSchemaRegistryConstants.PROXY_URL);
-    		try {
-    			this.proxyUrl = URI.create(value);
-    		} catch (IllegalArgumentException e) {
-        		String message = String.format("Proxy URL property is not a valid URL: %s", value);
-        		throw new AWSSchemaRegistryException(message, e);
-        	}
+            String value = (String) configs.get(AWSSchemaRegistryConstants.PROXY_URL);
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:227]: remove these spacing change please

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java [hunk 2]
```diff
@@ -216,4 +233,4 @@ public enum COMPRESSION {
          */
         ZLIB
     }
-}
+}
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java:236]: unnecessary formatting change, please remove

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+     * Constructor used by Kafka Connect user.
+     */
+    public CrossRegionReplicationMM2Converter(){};
+
+    /**
+     * Constructor accepting AWSCredentialsProvider.
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    public CrossRegionReplicationMM2Converter(
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:47]: this constructor is only used for testing purpose right? lets convert it to package-private

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+        new CrossRegionReplicationMM2ConverterConfig(configs);
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:69]: why the value is initialized but not assigned?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            return encodedByte;
+
+        } catch (SerializationException | AWSSchemaRegistryException e){
+            throw new DataException("Converting Kafka Connect data to byte[] failed due to serialization/deserialization error: ", e);
+        }
+
+    }
+
+    @Override
+    public SchemaAndValue toConnectData(String s, byte[] bytes) {
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:129]: Let us add comment here stating this method should never be invoked since MM2 is a source connector. In addition to that, let us throw an exception instead of returning null

File: cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for testing RegisterSchema class.
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+
+public class CrossRegionReplicationMM2ConverterTest {
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java:42]: can we also have some unit tests that covers Json schema and protobuf schema?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+import com.amazonaws.services.schemaregistry.common.Schema;
+import com.amazonaws.services.schemaregistry.common.configs.GlueSchemaRegistryConfiguration;
+import com.amazonaws.services.schemaregistry.deserializers.GlueSchemaRegistryDeserializationFacade;
+import com.amazonaws.services.schemaregistry.deserializers.GlueSchemaRegistryDeserializerImpl;
+import com.amazonaws.services.schemaregistry.exception.AWSSchemaRegistryException;
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializationFacade;
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializerImpl;
+import com.amazonaws.services.schemaregistry.utils.AWSSchemaRegistryConstants;
+
+import kotlinx.serialization.SerializationException;
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:13]: Is this the correct exception to use?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    private GlueSchemaRegistrySerializationFacade serializationFacade;
+    private AwsCredentialsProvider credentialsProvider;
+    private GlueSchemaRegistryDeserializerImpl deserializer;
+    private GlueSchemaRegistrySerializerImpl serializer;
+    private boolean isKey;
+
+    /**
+     * Constructor used by Kafka Connect user.
+     */
+    CrossRegionReplicationMM2Converter(){};
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:40]: Shouldn't these constructors be public?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+import org.apache.kafka.connect.storage.Converter;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Data
+public class CrossRegionReplicationMM2Converter implements Converter {
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:28]: Good to add class level documentation on how this works. 

We can remove references to MM2 as I don't see anything specific to MM2 here. This can be used with non-MM2 like connectors that need cross-region schemas. Please rename the class and update documentation accordingly.

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+        this.deserializationFacade = deserializationFacade;
+        this.serializationFacade = serializationFacade;
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:62]: Not necessarily specific to MM2. We can remove the reference to MM2.

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:70]: We should be able to support credentialProvider passed on by the user. Please add a TODO to take it up separately.

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+                GlueSchemaRegistryDeserializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(sourceConfigs)
+                        .build();
+
+        serializationFacade   =
+                GlueSchemaRegistrySerializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(targetConfigs)
+                        .build();
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:97]: These facades are low level classes and good to avoid using them if possible. Can we just use the `GlueSchemaRegistrySerializerImpl` and `GlueSchemaRegistryDeserializerImpl` classes?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+        serializer   = new GlueSchemaRegistrySerializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(targetConfigs));
+
+        deserializer = new GlueSchemaRegistryDeserializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(sourceConfigs));
+
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        byte[] bytes = (byte[]) value;
+        if (value == null) return new byte[0];
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:108]: If value is null, we cannot return empty bytes as empty bytes are different from null bytes. This can also be the first statement so that we don't risk processing nulls.

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+
+        try {
+            byte[] deserializedBytes = deserializer.getData(bytes);
+            Schema deserializedSchema = deserializer.getSchema(bytes);
+
+            byte[] encodedByte = serializer.encode(null, deserializedSchema, deserializedBytes);
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:114]: Instead of passing transportName null, please check how we are passing it in other cases. Can it be derived from topic name?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+            com.amazonaws.services.schemaregistry.common.Schema returnedSchema = getSchema(bytes);
+            UUID uuid = registerSchema(returnedSchema);
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:116]: Isn't this automatically done by the `serializer`? [Ref](https://github.com/awslabs/aws-glue-schema-registry/blob/master/serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/serializers/GlueSchemaRegistrySerializationFacade.java#L112) Why do we have to do it again?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+
+            return encodedByte;
+
+        } catch (SerializationException | AWSSchemaRegistryException e){
+            throw new DataException("Converting Kafka Connect data to byte[] failed due to serialization/deserialization error: ", e);
+        }
+
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:123]: Minor: remove

Please fix the newlines everywhere.

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+    /**
+     * This method is not intended to be used for the CrossRegionReplicationMM2Converter given that MM2 is a source connector
+     *
+     */
+    @Override
+    public SchemaAndValue toConnectData(String topic, byte[] value) {
+
+        try {
+            throw new IllegalAccessException("This method is not intended to be used here");
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:134]: Maybe throw `UnsupportedOperationException`?

File: cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java (hunk 6 above)
Comment (blacktooth) [reply cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java:42]: +1 

We can have 1 test per data format. That test can use a complex schema so that it covers all the cases.

File: cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java [hunk 18]
```diff
@@ -0,0 +1,400 @@
...
+    byte[] genericBytes = new byte[] {3, 0, -73, -76, -89, -16, -100, -106, 78, 74, -90, -121, -5,
+            93, -23, -17, 12, 99, 10, 115, 97, 110, 115, 97, -58, 1, 6, 114, 101, 100};
+
+
+    byte[] avroBytes = new byte[] {3, 0, 84, 24, 47, -109, 37, 124, 74, 77, -100,
+            -98, -12, 118, 41, 32, 57, -66, 30, 101, 110, 116, 101, 114, 116, 97, 105, 110, 109, 101, 110,
+            116, 95, 50, 0, 0, 0, 0, 0, 0, 20, 64};
+
+    private CrossRegionReplicationMM2Converter converter;
+
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java:67]: Minor: please remove extra new lines here

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+        }
+            return deserializationFacade.getSchema(data);
+    }
+
+    /**
+     * Register schema in the target region GSR
+     * @param schema schema obtained from the source region GSR
+     * @return schema version ID of the registered schema
+     */
+    public  UUID registerSchema(com.amazonaws.services.schemaregistry.common.Schema schema){
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:158]: How is the registry chosen in the target region?  Also, how is the schema named?

File: cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java [hunk 20]
```diff
@@ -0,0 +1,400 @@
...
+    @BeforeEach
+    void setUp() {
+        converter = new CrossRegionReplicationMM2Converter(deserializationFacade, serializationFacade, credProvider, deserializer, serializer);
+    }
+
+    /**
+     * Test for Mm2Converter config method.
+     */
+    @Test
+    public void testConverter_configure() {
```
Comment (blacktooth) [inline cross-region-replication-mm2-converter/src/test/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2ConverterTest.java:79]: I didn't find a test that tests the public method `convert`. Am I missing something?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 13 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:108]: Does it mean I should return null or null bytes (I was looking null bytes up but not so certain which one to 	use)

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 21]
```diff
@@ -44,6 +44,10 @@ public class GlueSchemaRegistryConfiguration {
     private AWSSchemaRegistryConstants.COMPRESSION compressionType = AWSSchemaRegistryConstants.COMPRESSION.NONE;
     private String endPoint;
     private String region;
+    private String sourceEndPoint;
```
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:48]: I had some trouble removing the configurations only for cross schema replication from the GlueSchemaRegistryConfiguration.file so I have temporally added as one of the TODOs

File: cross-region-replication-mm2-converter/pom.xml [hunk 22]
```diff
@@ -0,0 +1,229 @@
...
+ * limitations under the License.
+ */
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>${parent.groupId}</groupId>
+    <artifactId>cross-region-schema-registry-kafkaconnect-converter</artifactId>
```
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/pom.xml]: Have changed the name of the artifacts

File: cross-region-replication-mm2-converter/pom.xml [hunk 23]
```diff
@@ -0,0 +1,229 @@
...
+    <version>${parent.version}</version>
+    <name>AWS Cross Region Glue Schema Registry Kafka Connect AVRO Converter</name>
```
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/pom.xml]: Have removed the AVRO part

File: cross-region-replication-mm2-converter/pom.xml [hunk 24]
```diff
@@ -0,0 +1,229 @@
...
+    <description>The AWS Glue Schema Registry Kafka Connect Converter enables Java developers to easily integrate
+        their Kafka Connect applications with AWS Glue Schema Registry
```
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/pom.xml]: Have changed the description

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 7 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: Have changed to class used for the exception

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 10 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: Have renamed excluding the MM2 part

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 19 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: This function is longer used

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 17 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: Changed exception to UnsupportedOperationException

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 16 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: deleted new lines

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 12 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: Have removed the usage of GlueSchemaRegistrySerializerImpl & GlueSchemaRegistryDeserializerImpl

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 11 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java]: Have add a TODO for supporting credentialProvider

File: cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+            GlueSchemaRegistryDeserializerImpl deserializerImpl,
+            GlueSchemaRegistrySerializerImpl serializerImpl) {
+
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
```
Comment (blacktooth) [inline cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:51]: Remove MM2?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (blacktooth) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:48]: Can you please create an issue in the backlog?

File: cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java (hunk 25 above)
Comment (JC-CJ2222) [reply cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:51]: Removed MM2

File: .gitignore [hunk 26]
```diff
@@ -10,4 +10,6 @@
 **/*.project
 **/*.classpath
 **/*.factorypath
-**/dependency-reduced-pom.xml
\ No newline at end of file
+**/dependency-reduced-pom.xml
+/integration-tests/logs/
+/integration-tests/kafka_2.13-3.4.0/
```
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply .gitignore:15]: Fixed

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:48]: Done!

**Comment by YangSan0622** [comment]
Can we not squash the commits to 1 commit?

File: cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        if (value == null) return null;
+        byte[] bytes = (byte[]) value;
+
+        try {
+            byte[] deserializedBytes = deserializer.getData(bytes);
+            Schema deserializedSchema = deserializer.getSchema(bytes);
+            //The registry is decided by the configuration in the target region , schema name is the same as the source region
+            return serializer.encode(topic, deserializedSchema, deserializedBytes);
```
Comment (YangSan0622) [inline cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:88]: are you going to work on prefixing topic name the source cluster alias so we can have the registered schema name same as topic name in target region?

Comment (YangSan0622) [reply cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:88]: If you will not work on it then please also create a issue in the backlog for us to keep track. Link the issue in TODO as well.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (YangSan0622) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:48]: Can you link the issue in TODO so others reading your code is aware of it?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 28]
```diff
@@ -186,15 +209,29 @@ private void validateAndSetAWSEndpoint(Map<String, ?> configs) {
...
     private void validateAndSetProxyUrl(Map<String, ?> configs) {
         if (isPresent(configs, AWSSchemaRegistryConstants.PROXY_URL)) {
-    		String value = (String) configs.get(AWSSchemaRegistryConstants.PROXY_URL);
-    		try {
-    			this.proxyUrl = URI.create(value);
-    		} catch (IllegalArgumentException e) {
-        		String message = String.format("Proxy URL property is not a valid URL: %s", value);
-        		throw new AWSSchemaRegistryException(message, e);
-        	}
+                String value = (String) configs.get(AWSSchemaRegistryConstants.PROXY_URL);
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:228]: are you still not able to fix the spacing issue here?

File: cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+     * Configure the Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+        //TODO: Support credentialProvider passed on by the user
```
Comment (YangSan0622) [inline cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:58]: Let us also create a issue here and link it in TODO for easy tracking

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (JC-CJ2222) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:49]: Done!

File: cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java (hunk 27 above)
Comment (JC-CJ2222) [reply cross-region-replication-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationConverter.java:91]: Done!
